[CS2113-T10-2] BookBob

kaboomzxc opened the pull request:
BookBob helps General Practitioner Doctors to manage patient details. It is optimised for CLI users, so that they are able to access 
information of the patients faster by searching their specific details such as Name or NRIC.

TJW0911 commented on src/main/java/bookbob/Main.java:
```diff
@@ -0,0 +1,254 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Main {
```
This main method is very long. Perhaps you could consider refactoring the code to separate the functionalities for parsing with a 'parser' package similar to what was done in the IP.

TJW0911 commented on src/main/java/bookbob/Main.java:
```diff
@@ -0,0 +1,254 @@
+                    logger.log(Level.WARNING, "Error processing add command", e);
+                    System.out.println("Error in adding patient record, specific error: " + e.getMessage());
+                }
+                break;
```
Perhaps it could be a good idea to extract the logic in each individual switch cases as a method as some of these switch cases are pretty lengthy and extracting method would make the code more concise and readable. This applies to all switch cases in the main method.

TJW0911 commented on src/main/java/bookbob/entity/AppointmentRecord.java:
```diff
@@ -0,0 +1,131 @@
+        List<Appointment> appointments = new ArrayList<>();
+        String[] inputs = input.split("/");
+        String details = inputs[1];
+        if (inputs[0].equals("n")) {
```
Maybe it would be a good idea to do 
String filter = inputs[0];
and then use the filter variable to check the conditions for easier readability.

TJW0911 commented on src/main/java/bookbob/functions/FileHandler.java:
```diff
@@ -0,0 +1,159 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FileHandler {
```
Maybe you could use slightly different method names to name the different methods in this file. I was confused for a bit when I saw multiple pairs of methods with the exact same method name.

TJW0911 commented on src/main/java/bookbob/Main.java:
```diff
@@ -0,0 +1,254 @@
+
+        while (true) {
+            String input = in.nextLine();
+            String[] inputArr = input.split(" ", 2);
```
Maybe it would be a good idea to name the variable in full as Arr is not that commonly used to refer to Array.

louisjoety commented on docs/NewAppointment.puml:
```diff
@@ -0,0 +1,29 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
```
Neat and tidy, very detailed.

You may consider adding colours and hide the footbox 👍

louisjoety replied:
I found many rounded rectangles instead of rectangles, from my knowledge, we have to use rectangles.

louisjoety commented on docs/AfterAppointment.puml:
```diff
@@ -0,0 +1,27 @@
+object "<u> :CommandHandler</u>" as CommandHandler
+object "<u> in:Scanner</u>" as Scanner
+object "<u> appointment:Appointment </u>" as Appointment {
+name = John Doe
+nric = S123A
+date = 18-11-2024
+time = 18:00
```
Can consider using "" for strings

louisjoety commented on docs/NewAppointment.puml:
```diff
@@ -0,0 +1,29 @@
+'https://plantuml.com/sequence-diagram
+
+actor Doctor
+Doctor -> Main ++
+Main -> Record **
+Main -> AppointmentRecord **
+Main -> FileHandler ++ : initFile(records)
+FileHandler -> Main -- :
+Main -> FileHandler ++ : initFile(appointmentRecords)
+FileHandler -> Main -- :
+Main -> CommandHandler ** :
+Doctor -> Main : appointment("John Doe, S123A, 18-11-2024, 18:00")
+Main -> CommandHandler ++: commandHandler.appointment("John Doe, S123A, 18-11-2024, 18:00")
+CommandHandler -> AppointmentRecord ++ : appointmentRecord.checkAvailability("18-11-2024, 18:00")
+AppointmentRecord -> CommandHandler -- : nextAvailableTime
+group alt [nextAvailableTime == "18:00"]
+    CommandHandler -> Appointment **
+    CommandHandler -> Appointment ++ : new Appointment("John Doe", "S123A", "18-11-2024", "18:00")
+    Appointment -> CommandHandler -- : appointment
+
+    CommandHandler -> AppointmentRecord ++ : appointmentRecords.addAppointment(appointment)
+    AppointmentRecord -> CommandHandler --
+    else
+end
+CommandHandler -> FileHandler ++ : FileHandler.autosave(appointmentRecords)
+FileHandler -> CommandHandler --
```
You may wish to consider deactivating the activation bar after the completion

louisjoety commented on docs/NewAppointment.puml:
```diff
@@ -0,0 +1,29 @@
+Doctor -> Main : appointment("John Doe, S123A, 18-11-2024, 18:00")
+Main -> CommandHandler ++: commandHandler.appointment("John Doe, S123A, 18-11-2024, 18:00")
+CommandHandler -> AppointmentRecord ++ : appointmentRecord.checkAvailability("18-11-2024, 18:00")
+AppointmentRecord -> CommandHandler -- : nextAvailableTime
```
I think for any return arrows, it should be `-->` instead

louisjoety commented on docs/AfterAppointment.puml:
```diff
@@ -0,0 +1,27 @@
+object "<u> record:Record </u>" as Record
+object "<u> :FileHandler </u>" as FileHandler
+
+Main --> Record
+Main --> AppointmentRecord
+Main --> FileHandler
+Main --> CommandHandler
+Main --> Scanner
+CommandHandler --> FileHandler
+CommandHandler --> AppointmentRecord
+CommandHandler --> Record
+AppointmentRecord --> Appointment
```
Object diagram well drawn, can consider adding labels to make it clearer

louisjoety reviewed (commented):
Keep it up! 😺

louisjoety commented on docs/NewAppointment.puml:
```diff
@@ -0,0 +1,29 @@
+'https://plantuml.com/sequence-diagram
+
+actor Doctor
+Doctor -> Main ++
+Main -> Record **
+Main -> AppointmentRecord **
+Main -> FileHandler ++ : initFile(records)
+FileHandler -> Main -- :
+Main -> FileHandler ++ : initFile(appointmentRecords)
+FileHandler -> Main -- :
+Main -> CommandHandler ** :
+Doctor -> Main : appointment("John Doe, S123A, 18-11-2024, 18:00")
+Main -> CommandHandler ++: commandHandler.appointment("John Doe, S123A, 18-11-2024, 18:00")
```
Missing colon in front of the entity name (i.e.":Main" instead of "Main" )

gvnchua commented on docs/AfterExecutionOD.png:
Is the before and after execution diagram supposed to be exactly the same?

anlamm commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+Object Diagram after execution of appointment feature:
+![img.png](AfterExecutionOD.png)
+
+The Sequence Diagram for the execution of the appointment feature:
+![img.png](NewAppointmentSD.png)
```
Shouldn't the arrows of creating object point to its activation bar?

cxc0418 replied:
would it be helpful to label key interactions between objects (e.g., CommandHandler, AppointmentRecord, FileHandler) to clarify their roles?

gvnchua commented on docs/NewAppointmentSD.png:
should the record object have a constructor?

gvnchua commented on docs/NewAppointmentSD.png:
same for Appointment record

gvnchua commented on docs/NewAppointmentSD.png:
Appointment should also have a constructor

anlamm commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+
+Object Diagram after execution of appointment feature:
+![img.png](AfterExecutionOD.png)
+
```
It will be clearer if a class diagram is included.

anlamm replied:
Isn't the object diagram before execution of application feature and after execution of application feature the same?

cxc0418 commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+| Action | Format | Example |
+|---|---|---|
+| Help | `help` | `help` |
+| Add | `add n/NAME ic/NRIC [p/PHONE_NUMBER] [d/DIAGNOSIS] [m/MEDICATION] [ha/HOME_ADDRESS] [dob/DATE_OF_BIRTH] [v/VISIT_DATE_TIME] [al/ALLERGY] [s/SEX] [mh/MEDICALHISTORY]` | `add n/James Ho ic/S9534567A p/91234567 d/Asthma m/Albuterol ha/NUS-PGPR dob/01011990 v/21-10-2024 15:48 al/Pollen s/Female mh/Diabetes` |
```
For the appointment command syntax, should the date format dd-mm-yyyy and time format HH:mm be clarified further in the command summary?

cxc0418 commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+
+### Visit Management
+
+1. Adding visits
+    1. (Positive) Test Case: `addVisit ic/S9876543A v/21-10-2024 15:48 d/Fever,Cough m/Paracetamol,Cough Syrup` <br>
+       Expected: Visit added to patient's record with diagnoses and medications.
```
Would providing a few additional examples for variations help reduce input errors?

cxc0418 commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+
+### Data Persistence (Saving and Loading)
+
+1. Automatic Storage
+    1. Test case: Add/edit/delete records, then restart application
+        - Expected: All changes are preserved after restart
+        - Note: Data is automatically saved to `bookbob_data.txt` in the `data` folder (same directory as Bookbob.jar)
+
+2. File Management
+    1. Test case: Manually View saved data
+        - Action: Navigate to `data` folder (same directory as BookBob.jar) and open `bookbob_data.txt`
+        - Note: Do not manually modify the file contents to prevent data corruption. Saving and loading data is automated as long as file is not corrupted.
 
-## Instructions for manual testing
+    2. Test case: Delete data file and restart application
+        - Action: Delete `bookbob_data.txt` from the `data` folder
+        - Expected: New `bookbob_data.txt` file is automatically generated. The missing text file will not result in any error as files will be generated automatically.
+        - Note: This can be used to start with a fresh database if needed
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+3. Error Handling
+    1. Test case: Corrupt the data file manually, then start application
+        - Expected: Error message is shown about invalid data
+        - Note: BookBob will continue to function
+        - Recovery options:
+            * Option 1: Manually remove corrupted lines from `bookbob_data.txt`
+            * Option 2: Delete `bookbob_data.txt` to start afresh
```
is it necessary to expand on recovery options when the data file is corrupted? Should we add specific steps or a short troubleshooting guide to assist users in recovering from data corruption errors?

gvnchua commented on docs/NewAppointmentSD.png:
Should the return arrows be dotted

anlamm commented on docs/DeveloperGuide.md:
```diff
@@ -25,14 +29,213 @@
+
+The Sequence Diagram for the execution of the appointment feature:
+![img.png](NewAppointmentSD.png)
+
```
Shouldn't the return arrow in dash line?

gvnchua commented on docs/NewAppointmentSD.png:
the activation bar for Main and CommandHandler are not closed

anlamm commented on docs/DeveloperGuide.md:
```diff
@@ -11,12 +11,16 @@
-{Describe the target user profile}
+Dr Bob is a General Practitioner running his own private clinic.
+He manages everything independently, attending to numerous patients with diverse health concerns each day.  
+The demanding workload and long hours often leave him exhausted and sleep-deprived.  
```
Maybe a bit too much detail about background?

gvnchua commented on docs/NewAppointmentSD.png:
CommandHandler should call autoSave in FileHandler after getting the return nextAvailableTime

mikolajed commented on docs/AddVisitSequenceDiagram.puml:
```diff
@@ -0,0 +1,75 @@
+
+Doctor -> ":Main" ++ : addVisit ic/S9870789B v/29-10-2024 10:00 d/Gastric m/Gaviscon, Paracetamol
+
+":Main" -> ":Scanner"**: new Scanner()
+activate ":Scanner"
+":Main" <-- ":Scanner": in:Scanner
+deactivate ":Scanner"
+
+":Main" -> ":Records"**: new Records()
+activate ":Records"
+":Main" <-- ":Records": records:Records
+deactivate ":Records"
+
+":Main" -> ":AppointmentRecord"**: new AppointmentRecord()
+activate ":AppointmentRecord"
+":Main" <-- ":AppointmentRecord": appointmentRecord:AppointmentRecord
+deactivate ":AppointmentRecord"
+
+participant ":FileHandler" as FileHandler <<class>>
+":Main" -> FileHandler ++: initFile(records)
+return
+
+":Main" -> FileHandler ++: initFile(appointmentRecord)
+return
```
There seems to be an overlap with the MainSequenceDiagram.puml. Moving this part to a separate diagram and just referencing it would improve readability, would you agree?

mikolajed commented on docs/CommandHandler.puml:
```diff
@@ -0,0 +1,50 @@
+class CommandHandler {
+-{static} logger: Logger
++CommandHandler(): void
++help(): void
++add(input: String, records: Records): void
++list(records: Records): void
++edit(input: String, records: Records): void
++editVisit(input: String, records: Records): void
++delete(nric: String, records: Records): void
++find(input: String, records: Records): void
++exit(input: String): void
++addVisit(input: String, records: Records): void
++Appointment(input: String, appointmentRecord: AppointmentRecord): void
++deleteAppointment(input: String, appointmentRecord: AppointmentRecord): void
++listAppointment(appointmentRecord: AppointmentRecord): void
++findAppointment(input: String, appointmentRecord: AppointmentRecord): void
++deleteAppointment(appointmentRecord: AppointmentRecord): void
+-findNextFieldStart(input: String, currentIndex: int): int
```
I like that it is clear which component is responsible for what. Wouldn't all of these methods be a bit overwhelming for a reader? Isn't it too detailed?

mikolajed commented on docs/Main.puml:
```diff
@@ -0,0 +1,25 @@
+@startuml
+'https://plantuml.com/class-diagram
+
+class Main {
+-{static} logger: Logger {readOnly}
++{static} main(args: String[]): void
+}
+
+class Patient {}
+class Visit {}
+class Main {}
+class Records {}
+class CommandHandler {}
+class AppointmentRecord {}
+
+
+CommandHandler --> Appointment : searches and modifies >
+AppointmentRecord "1" o-- "many" Appointment
+Main --> CommandHandler : calls >
+Main --> FindVisit : calls >
+CommandHandler --> Patient : searches and modifies >
+FindVisit --> Visit : searches >
+Records "1" o-- "many" Patient
+Patient "1" o-- "many" Visit
+@enduml
```
I really like this diagram is provided, important for understanding other ones.

mikolajed commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,353 @@
+**Doctor enters the command:** `add n/Patricia Chan ic/S9870789B p/98097890 d/Cough m/Antibiotics ha/Bukit Batok East Avenue 3 
+dob/01111998 v/21-10-2024 18:00 al/Peanuts s/Female mh/Hypertension`
+
+> The Object Diagram before the execution of "add" command:
+![img.png](ObjectDiagramBeforeAddPatient.png)
+
+> The Object Diagram after the execution of "add" command:
+![img.png](ObjectDiagramAfterAddPatient.png)
```
It is not blatantly clear what is being changed, too much detail. Maybe it would be beneficial to put the diagrams side by side?

mikolajed reviewed (commented):
In general, the diagrams follow the confection with no major violations, good job. However, there seems to be too much detail in the DeveloperGuide.md as well as on the diagrams themselves.

Commit:
add author

kaboomzxc committed:
Merge pull request #287 from kaboomzxc/master

add author

Commit:
fix author

kaboomzxc committed:
Merge pull request #288 from kaboomzxc/master

fix author

yentheng0110 committed:
Change the help output to guide the users correctly

yentheng0110 committed:
Make the  error message printed out for find command consistent for all optional parameters (denoted by square brackets)

kaboomzxc committed:
Merge pull request #290 from yentheng0110/YenTheng-FinalCheckBeforeReleasingV2.0

Do a final check before releasing v2.0

G13nd0n committed:
Fix bugs in the commands

G13nd0n committed:
Fix typo issue

G13nd0n committed:
Fixed typo

yentheng0110 committed:
Merge pull request #292 from G13nd0n/master

Fix bugs in the commands

yentheng0110 committed:
Fix minor typo in Main.java

kaboomzxc committed:
Merge pull request #293 from yentheng0110/master

Fix minor typo in Main.java

G13nd0n committed:
Update the DeveloperGuide based on feedback received

G13nd0n committed:
Extract certain variables used throughout the tests

coraleaf0602 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-T10-2/tp

* 'master' of https://github.com/AY2425S1-CS2113-T10-2/tp: (58 commits)
  Fix minor typo in Main.java
  Fixed typo
  Fix typo issue
  Fix bugs in the commands
  Make the  error message printed out for find command consistent for all optional parameters (denoted by square brackets)
  Change the help output to guide the users correctly
  fix author
  add author
  add author
  update_DG
  fix CI
  Main Class Test
  Create RecordsTest class with JUnit tests to ensure 100% test coverage (class, method, line, branch)
  Update author tags
  Delete BookBobTest.java from v1.0 (we shifted all test methods into CommandHandlerTest.java)
  Create a PatientTest class and add JUnit tests to achieve 100% test coverage (class, method, line and branch)
  Update AboutUs.md
  Update AboutUs.md
  Update princecatt.md
  Update AboutUs.md
  ...

# Conflicts:
#	docs/AboutUs.md

coraleaf0602 committed:
Add interface for file operations in FileHandler

coraleaf0602 committed:
Add interface for converting file output

coraleaf0602 committed:
Modify Patient class to include OutputConversion interface

coraleaf0602 committed:
Modify Patient class to include OutputConversion interface

coraleaf0602 committed:
Modify AppointmentRecord class to include FileOperation interface

coraleaf0602 committed:
Modify Record class to include FileOperation interface

coraleaf0602 committed:
Remove unused imports and refactored code

G13nd0n committed:
Modify the class files to extract duplicate methods, shift delete and add to under Records and shift deleteAppointment to AppointmentRecord

G13nd0n committed:
Modify attribute of Appointment

yentheng0110 committed:
Merge pull request #343 from G13nd0n/master

Refine the code to have lesser code duplication

coraleaf0602 committed:
Import FileHandler class to Records class to use static methods

coraleaf0602 committed:
Update authorship for methods

coraleaf0602 committed:
Delete commented code from FileHandler

coraleaf0602 committed:
Merge branch 'master' into Inheritance-File-Handling

coraleaf0602 committed:
Change test method to adhere to code structure in FileHandler

coraleaf0602 committed:
Merge branch 'Inheritance-File-Handling' of https://github.com/coraleaf0602/tp into Inheritance-File-Handling

* 'Inheritance-File-Handling' of https://github.com/coraleaf0602/tp:
  Modify attribute of Appointment
  Modify the class files to extract duplicate methods, shift delete and add to under Records and shift deleteAppointment to AppointmentRecord
  Extract certain variables used throughout the tests
  Update the DeveloperGuide based on feedback received

coraleaf0602 committed:
Bug Fix: Missing semicolon

yentheng0110 committed:
Merge pull request #342 from coraleaf0602/Inheritance-File-Handling

Refactoring FileHandler

PrinceCatt committed:
before_pull

PrinceCatt committed:
Merge branch 'master' of https://github.com/PrinceCatt/tp
Refactor Main and write log into a file

yentheng0110 committed:
Merge pull request #344 from PrinceCatt/master

Refactor Main and Logging

G13nd0n committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-T10-2/tp

G13nd0n committed:
Fix bugs in the add function

G13nd0n committed:
Fix minor bug in the removePastAppointment Function

PrinceCatt committed:
Convert DOB to localdate and change corresponding methods
Fix caused bugs in tests

yentheng0110 committed:
Merge pull request #345 from G13nd0n/master

Fix bugs in the add function

PrinceCatt committed:
fix_CI

PrinceCatt committed:
Merge branch 'master' into master

PrinceCatt committed:
fix_CI2

PrinceCatt committed:
Merge branch 'master' of https://github.com/PrinceCatt/tp

PrinceCatt committed:
fix_CI3

PrinceCatt committed:
minor_change

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
Update CommandHandler.java

PrinceCatt committed:
fix_CI

PrinceCatt committed:
merge

PrinceCatt committed:
Add_authorship

PrinceCatt committed:
add space between if and (

yentheng0110 committed:
Merge pull request #346 from PrinceCatt/master

Convert DOB to localdate and change corresponding methods

yentheng0110 committed:
Create a Parser class for more OOP

PrinceCatt committed:
Merge pull request #348 from yentheng0110/YenTheng-CreateParserClass

Create a Parser class for more OOP

PrinceCatt committed:
ban | in input

yentheng0110 committed:
Edit UG for add command such that DOB inputted is of the correct DD-MM-YYYY format

yentheng0110 committed:
Fix the bug so that when multiple mandatory fields are not provided, all error messages are shown to guide the user

yentheng0110 committed:
Fix the bug that name input is not given as the first input but name and other details are still extracted correctly

yentheng0110 committed:
Fix the bug for edit command, when the user forgets to provide updated fields, the patient records will not be deleted

yentheng0110 committed:
Fix the bug that home addresses with space and hyphens are accepted as valid

yentheng0110 committed:
Make sure phone number is not a mandatory field, a user won't be prompted to input a valid local phone number when the user did not input phone number at all

yentheng0110 committed:
Make sure that sex is not a mandatory field, a user is not prompted to input sex when the user did not

yentheng0110 committed:
Edit CommandHandler.java, adding assertions, fixing logical bugs and correcting formatting issues

yentheng0110 committed:
Update the Parser.java to delegate the validation of parameters to CommandHandler.java, allowing for the addition of assertions

yentheng0110 committed:
Commit the log file

yentheng0110 committed:
Delete MainTest.java as JunZheng will be implementing a different one

yentheng0110 committed:
Fix bug to accept spaces, hyphens, commas and slashes for names to be valid

yentheng0110 committed:
Modify test methods in CommandHandler.java to pass all tests

yentheng0110 committed:
Fix checkstyleMain issue

yentheng0110 committed:
Close issue #300 raised by peer tester

yentheng0110 committed:
Merge pull request #349 from PrinceCatt/master

ban | in input

coraleaf0602 committed:
Merge pull request #356 from yentheng0110/master

Fix bugs for BookBob

coraleaf0602 committed:
Add formatting for phone number

kaboomzxc committed:
Merge pull request #357 from coraleaf0602/Format-Phone-Number

Add formatting for phone number

Commit:
Main Sequence Diagram

Commit:
Object Diagram on initialisation of BookBob

Commit:
Update Main Class Diagram

Commit:
Update CommandHandler Class Diagram

Commit:
Updated FileHandler

Commit:
Update Records Class Diagram

Commit:
update AppointmentRecord Class Diagram

Commit:
update Appointment Class Diagram

Commit:
update Visit Class Diagram

Commit:
update Patient Class Diagram

Commit:
update ObjectDiagramBeforeAddVisit object diagram

Commit:
Update Object Diagram After Add Visit

Commit:
updated AddVisitSequenceDiagram

Commit:
update ObjectDiagramBeforeAddPatient Object Diagram

Commit:
Update ObjectDiagramAfterAddPatient  object diagram

Commit:
Update New Patient Sequence Diagram

G13nd0n committed:
Fix bugs based on feedbacks given and update UG

Commit:
Update Object Diagram for Before Appointment and After Appointment ,
and Sequence Diagram for "appointment" command

G13nd0n committed:
Fix CI issue

G13nd0n committed:
Fix CI and checkstyle issue

G13nd0n committed:
Update both DG and UG for the amendments made

Commit:
update UG

G13nd0n committed:
Update findAppointment to be case insensitive

kaboomzxc committed:
Merge pull request #358 from G13nd0n/master

Fix bugs based on feedbacks given and update UG

G13nd0n committed:
Merge pull request #359 from kaboomzxc/Class&Object&Seq_Diagrams_10Nov_(1)

update Class&Object&Seq Diagrams

G13nd0n committed:
Merge pull request #360 from kaboomzxc/update_UG_10Nov_(1)

Update UG

Commit:
Update UG

Commit:
update UG

Commit:
update UG typo

PrinceCatt committed:
Make find visit by diagnosis and by medication case insensitive

Commit:
update UG

kaboomzxc committed:
Merge pull request #362 from kaboomzxc/update_UG_10Nov_(2)

Update UG

kaboomzxc committed:
Merge pull request #363 from PrinceCatt/master

Make find visit by diagnosis and by medication case insensitive

Commit:
update UG

kaboomzxc committed:
Merge pull request #364 from kaboomzxc/master

update UG

PrinceCatt committed:
Add error report and auto-exiting when visit dates and DOB corrupted

PrinceCatt committed:
Include phone number and NRIC checks

PrinceCatt committed:
Include appointment record checks

PrinceCatt committed:
fix_CI

kaboomzxc committed:
Merge pull request #365 from PrinceCatt/master

Add error report and auto-exiting when data file corrupted

G13nd0n committed:
Fix bug of adding duplicate patient records

G13nd0n committed:
Modify the deleteAppointment to allow the deletion of appointment regardless of case-sensitivity

G13nd0n committed:
Fix bug of list command

yentheng0110 committed:
Merge pull request #366 from G13nd0n/master

Fix bugs

Commit:
update UG

Commit:
update ug

kaboomzxc committed:
Merge pull request #367 from kaboomzxc/update_UG_11Nov_(1)

update UG

Commit:
update UG

kaboomzxc committed:
Merge pull request #368 from kaboomzxc/update_UG_11Nov_(1)

update UG

Commit:
successfully solved :
guard against duplicate visits for "addVisit" command

Commit:
fix CI

Commit:
fix ci

yentheng0110 committed:
Merge pull request #369 from kaboomzxc/addVisit_NoDuplicate_Date&Time_(2)

guard against duplicate visits for "addVisit" command

yentheng0110 committed:
Modify Records.java to address many bug reports about adding patients with the same NRIC (ban the addition of duplicate NRIC)

yentheng0110 committed:
Add author tag

kaboomzxc committed:
Merge pull request #370 from yentheng0110/YenTheng-BanTheAdditionOfDuplicateNRIC

Modify Records.java to address many bug reports about adding patients with the same NRIC (ban the addition of duplicate NRIC)

yentheng0110 committed:
Modify DG to guide the users correctly

yentheng0110 committed:
Modify findVisit to include patient's name when printing out (besides visit details)

yentheng0110 committed:
Extract the method as printUnknownCommand(String command)

yentheng0110 committed:
Remove log file

PrinceCatt committed:
Fix a bug in checking for phone number corruption

kaboomzxc committed:
Merge pull request #372 from PrinceCatt/master

Fix a bug in checking for phone number corruption

yentheng0110 committed:
Modify UG to guide users correctly

yentheng0110 committed:
Add an extra space between } and catch for better readability

yentheng0110 committed:
Merge branch 'master' into YenTheng-AddressBugReports

yentheng0110 committed:
Bold add and addVisit command so that users can differentiate between the two

yentheng0110 committed:
Merge branch 'YenTheng-AddressBugReports' of https://github.com/yentheng0110/tp into YenTheng-AddressBugReports

yentheng0110 committed:
Modify UG for numbering issues and minor formatting issues

kaboomzxc committed:
Merge pull request #371 from yentheng0110/YenTheng-AddressBugReports

Modify UG, findVisit to include patient's name, extract the printing of unknown commands in Parser.java

Commit:
update UG

kaboomzxc committed:
Merge pull request #373 from kaboomzxc/11Nov330pm

update UG

Commit:
51 tests case passed,
100% class coverage, 100% method coverage, 69% line coverage, 89% branch coverage.

Commit:
removed unused imports and add author

Commit:
fix CI

Commit:
update name

Commit:
changed correct name

yentheng0110 committed:
Merge pull request #375 from kaboomzxc/ParserTest_11NovEvening_(2)

ParserTest

Commit:
"addVisit" command prevents addition of future dates.
Thank you Glendon for the helpful suggestion

Commit:
fix ci

Commit:
fix ci

Commit:
fix ci indentation

coraleaf0602 committed:
Merge pull request #376 from kaboomzxc/addVisit_NoFutureDate

"addVisit" command prevents addition of future dates.

G13nd0n committed:
Update UG and DG based on the refinement of methods

Commit:
"add" command prevents addition of future dates.

G13nd0n committed:
Fix issue of diagram for DG

Commit:
fix CI

Commit:
fix ci

kaboomzxc committed:
Merge pull request #377 from G13nd0n/master

Update UG and DG based on the refinement of methods

kaboomzxc committed:
Merge pull request #379 from kaboomzxc/add_Command_NoFutureDate

"add" command prevents addition of future dates.

Commit:
make diagrams bigger

kaboomzxc committed:
Merge pull request #380 from kaboomzxc/tryMakeDiagramsBigger

make diagrams bigger

yentheng0110 committed:
Change DG formatting issues for better readability

yentheng0110 committed:
Modify the command summary reference as there are some incorrect guidance

kaboomzxc committed:
Merge pull request #382 from yentheng0110/YenTheng-AddressBugReports

Modify DG's Command Summary Reference as there are some incorrect guidance, fix formatting issues for better readability

yentheng0110 committed:
Modify Table of Contents to make sure all links are clickable and Command Summary Reference are correctly placed at the end of the table

kaboomzxc committed:
Merge pull request #383 from yentheng0110/YenTheng-AddressBugReports

Modify Table of Contents to make sure all links are clickable and Command Summary Reference are correctly placed at the end of the table

yentheng0110 committed:
Fix formatting issues in DG

Commit:
try fix broken section formatting

kaboomzxc committed:
Merge pull request #384 from kaboomzxc/master

try fix broken section formatting

Commit:
fix DG formatting

kaboomzxc committed:
Merge pull request #386 from kaboomzxc/master

fix DG formatting

Commit:
update UG, DG

kaboomzxc committed:
Merge pull request #387 from kaboomzxc/fixUG&DG_12Nov130am

update UG, DG

Commit:
fix "find" finding DOB,
previously "find" suddenly cannot find DOB

Commit:
fix CI

Commit:
fix ci

yentheng0110 committed:
Update PPP

yentheng0110 committed:
Merge branch 'master' into YenTheng-AddressBugReports

yentheng0110 committed:
Merge pull request #390 from yentheng0110/YenTheng-AddressBugReports

Update PPP

Commit:
fix ci

yentheng0110 committed:
Fix formatting issues for PPP

yentheng0110 committed:
Merge pull request #391 from yentheng0110/master

Fix formatting issues for PPP

Commit:
fix ci

yentheng0110 committed:
Fix issues in formatting for PPP

Commit:
fix ci

yentheng0110 committed:
Merge pull request #392 from yentheng0110/master

Fix issues in formatting for PPP

Commit:
fix ci

Commit:
fix ci

Commit:
fix ci

kaboomzxc committed:
Merge pull request #388 from kaboomzxc/fix_find_dob

fix "find" finding DOB

yentheng0110 committed:
Shorten summary of contributions

yentheng0110 committed:
Fix minor formatting issues in PPP

yentheng0110 committed:
Merge pull request #393 from yentheng0110/master

Shorten summary of contributions in PPP

Commit:
update class diagrams Record and AppointmentRecord

yentheng0110 committed:
Final check for PPP

kaboomzxc committed:
Merge pull request #394 from kaboomzxc/master

update class diagrams Record and AppointmentRecord

yentheng0110 committed:
Merge pull request #395 from yentheng0110/YenTheng-FinalCheckForUG

Final check for PPP

yentheng0110 committed:
Fix help command and command summary in UG to guide the users correctly

yentheng0110 committed:
Shorten Overview of PPP

yentheng0110 committed:
Merge pull request #396 from yentheng0110/YenTheng-FinalCheckForUG

Shorten Overview of PPP

Commit:
JunZheng PPP

kaboomzxc committed:
Merge pull request #398 from kaboomzxc/JunZheng_PPP_(1)

JunZheng PPP

Commit:
update JZ PPP

kaboomzxc committed:
Merge pull request #399 from kaboomzxc/master

update JZ PPP

Commit:
fix typo JZ PPP

Commit:
update Main Object Diagram

kaboomzxc committed:
Merge pull request #400 from kaboomzxc/master

Update Main class diagram and fix typo in Jz PPP

Commit:
"patient record management" header in DG reduce size

kaboomzxc committed:
Merge pull request #401 from kaboomzxc/master

"patient record management" header in DG reduce size

yentheng0110 committed:
Modify UG for add command to include input validation section, explanations about mandatory fields and optional fields

yentheng0110 committed:
Modify extractGender(String input) in CommandHandler.java such that only "F", "Female", "M" or "Male" (case-insensitive) are allowed as inputs for sex

yentheng0110 committed:
Merge pull request #402 from yentheng0110/YenTheng-FinalCheckBeforeReleasingv2.1

Final check before releasing v2.1

yentheng0110 committed:
Fix minor formatting issues in UG

yentheng0110 committed:
Merge pull request #403 from yentheng0110/master

Fix minor formatting issues in UG

G13nd0n committed:
Update PPP

Commit:
update JZ PPP

kaboomzxc committed:
Merge pull request #406 from kaboomzxc/master

update JZ PPP

Commit:
update JZ PPP

kaboomzxc committed:
Merge pull request #407 from kaboomzxc/master

update JZ PPP

G13nd0n committed:
Update the enhancement contributed

G13nd0n committed:
Merge pull request #405 from G13nd0n/master

Update PPP

Commit:
update JZ PPP

kaboomzxc committed:
Merge pull request #408 from kaboomzxc/master

update JZ PPP

coraleaf0602 committed:
Update PPP

coraleaf0602 committed:
Update PPP

coraleaf0602 committed:
Merge pull request #409 from coraleaf0602/master

Update PPP for Cora Zhang

coraleaf0602 committed:
Update coraleaf0602.md

Commit:
fix Typos in UG

Commit:
fix CI

Commit:
fix CI

Commit:
fix CI

kaboomzxc committed:
Merge pull request #410 from kaboomzxc/master

fix Typos in UG

Commit:
update JZ PPP

kaboomzxc committed:
Merge pull request #411 from kaboomzxc/master

update JZ PPP
